src/pages/index.js

```diff
@@ -1,26 +1,40 @@
 import "./index.css";
 import PopupWithForm from "../components/PopupWithForm";
 import PopupWithImage from "../components/PopupWithImage";
+import PopupWithConfirm from "../components/PopupWithConfirm";
 import Section from "../components/Section";
 import Card from "../components/Card.js";
 import FormValidation from "../components/FormValidation.js";
-import { initCards, formData } from "../utils/constants.js";
+import { formData } from "../utils/constants.js";
 import UserInfo from "../components/UserInfo";
+import Api from "../components/Api";
 
 //search DOM
 //profile constants
-const profile = document.querySelector(".profile");
+const profile = document.querySelector(".page__profile");
 const editButton = profile.querySelector(".profile__button_type_edit");
 const cardButton = profile.querySelector(".profile__button_type_add");
+const avatarButton = profile.querySelector(".profile__button_type_avatar");
 
+//template constant
 const templateElement = document.querySelector(".template");
 
-// functions
+// utility functions
 const renderCard = (item) => {
   const card = new Card(
     item,
-    ({ title, link }) => {
-      previewPopup.open(title, link);
+    (title, link) => {
+      previewPopup.open({ name: title, link: link });
+    },
+    () => {
+      confirmPopup.open({ deleter: card.deleteCard, _id: card.getInfo()._id });
+    },
+    ({ isLiked, _id }) => {
+      if (isLiked) {
+        api.likeCard(_id).then(() => card.likeCard);
+      } else {
+        api.dislikeCard(_id).then(() => card.dislikeCard);
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**